scipy/optimize/optimize.py

```diff
@@ -183,6 +183,20 @@ def rosen(x):
     See Also
     --------
     rosen_der, rosen_hess, rosen_hess_prod
+    
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from scipy.optimize import rosen
+    
+    First, we need to create 1-D array to provide to Rosenbrock function::
+    
+    >>> X = np.random.randn(10)
+    
+    Now, We calculate the value of Rosenbrock function:
+    
+    >>> y = rosen(X)
+    >>> y
```

Comment:
The doctest actually checks the output of the function hence you have to provide the result. But random vector will keep changing its results. It's better to fix `X` and typeout the result explicitly 

```
>>> from scipy.optimize import rosen
>>> X = 0.1 * np.arange(10)
>>> rosen(X)
76.56
```